src/Agent.Worker/TestResults/TestDataPublisher.cs

```diff
@@ -83,21 +99,39 @@ private TestDataProvider ParseTestResultsFile(TestRunContext runContext, List<st
             return _parser.ParseTestResultFiles(_executionContext, runContext, testResultFiles);
         }
 
-        private bool GetTestRunOutcome(IList<TestRunData> testRunDataList)
+        private bool GetTestRunOutcome(IExecutionContext executionContext, IList<TestRunData> testRunDataList, out TestRunSummary testRunSummary)
         {
+            bool anyFailedTests = false;
+            testRunSummary = new TestRunSummary();
             foreach (var testRunData in testRunDataList)
             {
                 foreach (var testCaseResult in testRunData.TestResults)
                 {
-                    // Return true if outcome is failed or aborted
-                    if (testCaseResult.Outcome == TestOutcome.Failed.ToString() || testCaseResult.Outcome == TestOutcome.Aborted.ToString())
+                    testRunSummary.Total += 1;
+                    Enum.TryParse(testCaseResult.Outcome, out TestOutcome outcome);
+                    switch(outcome)
                     {
-                        return true;
+                        case TestOutcome.Failed:
```

Comment:
this switch statement feels like duplicate code. Is there any reason it cannot be refactored so you don't have duplicate logic?